auto_round/inference/convert_model.py

```diff
@@ -284,17 +304,14 @@ def get_layer_config(model, quantization_config):
     layer_names = list(set(layer_names).union(extra_config.keys()))
 
     # Construct final layer configuration
-    layer_configs = {
-        layer_name: {
-            "bits": extra_config.get(layer_name, {}).get("bits", bits),
-            "group_size": extra_config.get(layer_name, {}).get("group_size", group_size),
-            "data_type": extra_config.get(layer_name, {}).get("data_type", data_type),
-            "sym": extra_config.get(layer_name, {}).get("sym", sym),
-            "clip": extra_config.get(layer_name, {}).get("clip", False),
-        }
-        for layer_name in layer_names
-    }
-
+    layer_configs = {}
+    quant_scheme_attrs = QuantizationScheme.get_attributes()
+    for layer_name in layer_names:
+        layer_config = {}
+        layer_extra_config = extra_config.get(layer_name, {})
+        for scheme_attr in quant_scheme_attrs:
+            layer_config[scheme_attr] = layer_extra_config.get(scheme_attr, getattr(default_quant_scheme, scheme_attr))
+        layer_configs[layer_name] = QuantizationScheme.from_dict(layer_config)
```

Comment:
Hi @n1ck-guo @WeiweiZhang1, I wrapped the `layer_config` dict as a `QuantizationScheme`, and propagated it to the backend check. Please take a look and let me know if I missed any corner cases. Thanks.